hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,373 @@
+                # expressions/format
+                file_path = node[param_name].evaluate()
+
+                if not file_path or not os.path.exists(file_path):
+                    # no file or file does not exist, nothing to do
+                    continue
+
+                self.logger.info(
+                    "Processing %s node: %s" % (node_type, node.name()))
+
+                # file exists, let the basic collector handle it
+                item = super(NukeSessionCollector, self)._collect_file(
+                    parent_item,
+                    file_path,
+                    frame_sequence=True
+                )
+
+                # the item has been created. update the display name to include
+                # the nuke node to make it clear to the user how it was
+                # collected within the current session.
+                item.name = "%s (%s)" % (item.name, node.name())
+
+    def collect_sg_writenodes(self, parent_item):
+        """
+        Collect any rendered sg write nodes in the session.
+
+        :param parent_item:  The parent item for any sg write nodes collected
+        """
+
+        publisher = self.parent
+        engine = publisher.engine
+
+        sg_writenode_app = engine.apps.get("tk-nuke-writenode")
+        if not sg_writenode_app:
+            self.logger.debug(
+                "The tk-nuke-writenode app is not installed. "
+                "Will not attempt to collect those nodes."
+            )
+            return
+
+        for node in sg_writenode_app.get_write_nodes():
+
+            # see if any frames have been rendered for this write node
+            rendered_files = sg_writenode_app.get_node_render_files(node)
+            if not rendered_files:
+                continue
+
+            # some files rendered, use first frame to get some publish item info
+            path = rendered_files[0]
+            item_info = super(NukeSessionCollector, self)._get_item_info(path)
+
+            # item_info will be for the single file. we'll update the type and
+            # display to represent a sequence. This is the same pattern used by
+            # the base collector for image sequences. We're not using the base
+            # collector to create the publish item though since we already have
+            # the sequence path, template knowledge provided by the
+            # tk-nuke-writenode app. The base collector makes some "zero config"
+            # assupmtions about the path that we don't need to make here.
+            item_type = "%s.sequence" % (item_info["item_type"],)
+            type_display = "%s Sequence" % (item_info["type_display"],)
+
+            # we'll publish the path with the frame/eye spec (%V, %04d)
+            publish_path = sg_writenode_app.get_node_render_path(node)
+
+            # construct publish name:
+            render_template = sg_writenode_app.get_node_render_template(node)
```

Comment:
Any chance this could be None? What happens if the template setting is empty in the config?